vignettes/sir_models.Rmd

```diff
@@ -0,0 +1,399 @@
+
+```{r}
+effectiveSize(mcmc1)
+1 - rejectionRate(mcmc1)
+```
+
+Clearly the acceptance rate is too low. Generally a target of 0.5, reducing to 0.234 as the number of parameters increases, [is thought to be optimal](https://dx.doi.org/10.1016/j.spa.2007.12.005). The proposal distribution needs to be modified, in this case reducing the variance of the jump sizes. One option to automate this process is to run a chain for a bit, as above, then use the covariance of the state as the proposal matrix:
+
+```{r}
+proposal_matrix <- cov(pmcmc_run$pars)
+mcmc_pars <- mcstate::pmcmc_parameters$new(list(beta = beta, gamma = gamma), proposal_matrix)
+proposal_matrix
+```
+
+Let's now run four independent chains with these proposals:
+
+```{r, fig.height=10, fig.width=14}
+pmcmc_tuned_run <-
+  mcstate::pmcmc(
+    mcmc_pars,
+    filter,
+    2000,
+    save_state = TRUE,
+    save_trajectories = TRUE,
+    progress = TRUE,
+    n_chains = 4
+  )
+
+mcmc2 <- coda::as.mcmc(cbind(pmcmc_tuned_run$probabilities, pmcmc_tuned_run$pars))
+
+summary(mcmc2)
+plot(mcmc2)
+```
+
+Looking again at the effective sample size and rejection rate, these chains were a lot more efficient, and have a much better effective sample size thanks for the four chains:
+
+```{r}
+effectiveSize(mcmc2)
+1 - rejectionRate(mcmc2)
+```
+
+## Running predictions
+
+Going back to the simple SIR model above, we can easily continue running the fitted model forward in time using the posterior estimated by `pmcmc()` to create a forecast past the end of the data:
+
+```{r fig.height=5, fig.width=7}
+forecast <- predict(processed_chains,
+                    steps = seq(400, 800, 4),
+                    n_threads = 4L,
+                    prepend_trajectories = TRUE,
+                    seed = processed_chains$predict$seed)
+
+# Sample 10 particles
+mini_forecast <- forecast$state[,sample.int(ncol(forecast$state), size = 10),]
+
+plot_particle_filter(mini_forecast,
+                     true_history,
+                     seq_len(200),
+                     obs_end = max(incidence$day))
+```
+
+Up to $t = 100$ is the nowcast, from $t = 100$ to $t = 200$ is the forecast.
+
+## Fitting to multiple datastreams
+
+The set up of the particle filter and observation function makes it easy to fit to multiple datastreams within a Bayesian framework. Let us model a disease where a proportion of infected cases $p_{\mathrm{death}} = 0.05$ die rather than recover by adding the following lines to the odin code:
+
+```{r pkg_dust_r, echo = FALSE, results = "asis"}
+r_output(readLines(file.path("deaths.R")))
+```
+
+
+```{r}
+gen_death <- odin.dust::odin_dust("deaths.R")
+```
+
+We can simulate some data to fit to, as before. We have also made the case data patchy, so it is missing on some days:
+
+```{r layout="l-body-outset", paged.print=TRUE }
+# Set up model
+dt <- 0.25
+death_data <- read.table("death_data.csv", header = TRUE, sep = ",")
+true_history_deaths <- readRDS("deaths_true_history.rds")
```

Comment:
All these rds files need saving in the same script as the .csv - can you save them with `saveRDS(..., version = 2)` so that we use a version of rds that works with older R?